### PR TITLE
Add Homebrew to install docs

### LINKED ISSUE #1779: Distribute OSX release as a Homebrew formula
As a devops type nerd, I want to be able to install and update PowerShell/OSX via the CLI, and also with my preferred packaging tool, namely Homebrew (http://brew.sh). 

Yes, the the OSX native installer packages (.pkg) can be installed via CLI (with the `installer` command), but it requires sudo. AFAICT, PowerShell is installed to /usr/local/bin and does not require sudo at all otherwise. 

Homebrew is basically the de-facto CLI package manager on OSX like choco.exe is on Windows. It does not require sudo for packages to be installed.

Instructions to create a formula: https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Formula-Cookbook.md

### PR DESCRIPTION
**Pull request by joeyaiello**
Fixes #1779 

It would be awesome if someone who actually has a Mac (like @JamesWTruher or @BrucePay) can validate these instructions before we merge them.

Also, it would be great to know how accurate [these directions](https://github.com/PowerShell/PowerShell/compare/master...joeyaiello:HomebrewDocs?expand=1#diff-cb58069b7222dfcd5130fd97bf93e6e1R397) currently are. The reference to .NET Core 1.0.0 makes me think it might be a little outdated.

### CODE REVIEW

File: docs/installation/linux.md [hunk 1]
```diff
@@ -352,9 +379,14 @@ sudo installer -pkg powershell-6.0.0-beta.1-osx.10.12-x64.pkg -target /
 
 ### Uninstallation - macOS 10.12
 
-PowerShell on MacOS must be removed manually.
+If you PowerShell with Homebrew, uninstallation is easy:
+
+```sh
+brew uninstall powershell
```
Comment (JamesWTruher) [inline docs/installation/linux.md]: i needed to do `brew cask uninstall powershell`

File: docs/installation/linux.md [hunk 2]
```diff
@@ -339,6 +339,33 @@ and is a single binary.
...
+brew tap caskroom/cask
+brew cask install powershell
+```
+
+When new versions of PowerShell are released,
+simply update Homebrew's formulae and upgrade PowerShell:
+
+```sh
+brew update
+brew upgrade powershell
```
Comment (JamesWTruher) [inline docs/installation/linux.md]: i don't think these are right - they're not working for me. I get an error message:
```
brew upgrade powershell
Error: No available formula with the name "powershell" 
```

Comment (joeyaiello) [inline docs/installation/linux.md]: How about `brew cask upgrade powershell`?

File: docs/installation/linux.md (hunk 1 above)
Comment (joeyaiello) [inline docs/installation/linux.md]: Fixed.

File: docs/installation/linux.md (hunk 2 above)
Comment (vors) [inline docs/installation/linux.md]:
```
brew cask reinstall powershell
```

File: docs/installation/linux.md [hunk 3]
```diff
@@ -339,6 +339,33 @@ and is a single binary.
...
+
+[Homebrew][brew] is the missing package manager for macOS.
+If the `brew` command is not found,
+you need to install Homebrew following [their instructions][brew].
+
+Once you've installed Homebrew, installing PowerShell is easy.
+Just install [Homebrew-Cask][cask] and then you can install PowerShell:
+
+```sh
+brew tap caskroom/cask
```
Comment (vors) [inline docs/installation/linux.md]: This line is installation instruction for cask.
It's not clear from the comment that these two lines correspond to two steps above.

I'd break it down into two parts.

File: docs/installation/linux.md [hunk 4]
```diff
@@ -352,9 +379,14 @@ sudo installer -pkg powershell-6.0.0-beta.1-osx.10.12-x64.pkg -target /
 
 ### Uninstallation - macOS 10.12
 
-PowerShell on MacOS must be removed manually.
+If you PowerShell with Homebrew, uninstallation is easy:
```
Comment (vors) [inline docs/installation/linux.md]: Maybe `If you PowerShell installed with Homebrew` ? Not an English expert, but this language feels too playful :)

File: docs/installation/linux.md [hunk 5]
```diff
@@ -363,28 +395,26 @@ sudo rm -rf /usr/local/bin/powershell /usr/local/microsoft/powershell
...
+
+On macOS, .NET Core requires Homebrew's OpenSSL
+because the "OpenSSL" system libraries on macOS are not OpenSSL,
+as Apple deprecated OpenSSL in favor of their own libraries.
+This requirement is not a hard requirement for all of PowerShell.
+However, most networking functions (such as `Invoke-WebRequest`)
+do require OpenSSL to work properly.
 
-Also install [Homebrew's OpenSSL][openssl]:
+The easiest fix is to install [Homebrew's OpenSSL][openssl]:
```
Comment (vors) [inline docs/installation/linux.md]: This is auto-included in powershell brew formula (it's a dependency I think)
Worth pointing out here that it's not required as a separate step with brew.

**Review by vors — CHANGES_REQUESTED** [review]

**Comment by mirichmo** [comment]
@joeyaiello Don't forget about this one

File: docs/installation/linux.md (hunk 4 above)
Comment (joeyaiello) [inline docs/installation/linux.md]: Hahahaha, I just accidentally a word.

File: docs/installation/linux.md (hunk 2 above)
Comment (joeyaiello) [inline docs/installation/linux.md]: Wait, `reinstall` is the upgrade command?

**Comment by joeyaiello** [comment]
@mirichmo I hadn't! I just didn't realize anyone had responded yet? It's hard for me to get this one right because I don't actually have a Mac....

File: docs/installation/linux.md (hunk 2 above)
Comment (Happycoil) [inline docs/installation/linux.md]: @joeyaiello For now at least, see https://github.com/caskroom/homebrew-cask/issues/29301

**Comment by mirichmo** [comment]
@vors - Are all your concerns addressed?

**Comment by joeyaiello** [comment]
@mirichmo sorry, this is on me. I've been meaning to grab the Macbook from @daxian-dbw, gotta do my validation real quick.

File: docs/installation/linux.md (hunk 2 above)
Comment (joeyaiello) [inline docs/installation/linux.md]: Oh, interesting...thanks for the pointer. Looks like that's the way to go now. I'm updating the docs accordingly.

**Comment by joeyaiello** [comment]
Okay, sorry for taking so long on this. @mirichmo: it should be good to go now. I validated all the instructions on a Macbook. :)

**Comment by mirichmo** [comment]
@vors Do you have any additional comments or concerns?

**Comment by jpogran** [comment]
FWIW, followed these instructions on macOS 10.12.5 with homebrew and cask and it worked for me

**Comment by joeyaiello** [comment]
@mirichmo I think I have enough approval (and have adequately addressed @vors' concerns).

**Pull request merged**